toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to initialize a CSV file or record a new timestamp
+// for shell scripts during the image-building process.
+
+package main
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app            = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName     = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName       = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName     = app.Flag("action-name", "The name of the current action.").Default("").String()
+	dirPath        = app.Flag("dir-path", "The folder that stores timestamp CSVs.").Required().ExistingDir() // currently must be absolute
+	logFile        = app.Flag("log-file", "Directory for log files").Required().ExistingFile()
+	initializeMode = "i"
+	recordMode     = "r"
+	validModes     = []string{initializeMode, recordMode}
+	mode           = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('i') or 'record'('r').").Required().Enum(validModes...)
+	completePath   string
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logFile, "trace")
+
+	// Construct the CSV path.
+	completePath = filepath.Join(*dirPath, *scriptName+".csv")
```

Comment:
The modes are constant values we don't want to change - let's make them this way.
Also, no need to globally define `completePath`. We keep all `app.*` declarations here to make it easier to see the arguments for this tool, but in general, we want to limit the scope of variables and constants as much as possible.

Please double-check my code - I'm writing this from GitHub.:)

```suggestion
const (
	initializeMode = "i"
	recordMode     = "r"
)

var (
	app            = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
	scriptName     = app.Flag("script-name", "The name of the current tool.").Required().String()
	stepName       = app.Flag("step-name", "The name of the current step.").Required().String()
	actionName     = app.Flag("action-name", "The name of the current action.").Default("").String()
	dirPath        = app.Flag("dir-path", "The folder that stores timestamp CSVs.").Required().ExistingDir() // currently must be absolute
	logFile        = app.Flag("log-file", "Directory for log files").Required().ExistingFile()
	validModes     = []string{initializeMode, recordMode}
	mode           = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('i') or 'record'('r').").Required().Enum(validModes...)
)

func main() {
	app.Version(exe.ToolkitVersion)
	kingpin.MustParse(app.Parse(os.Args[1:]))
	logger.InitBestEffort(*logFile, "trace")

	// Construct the CSV path.
	completePath := filepath.Join(*dirPath, *scriptName+".csv")
```